### DISCUSSION TITLE
Point release 1.4.1 for lxqt-qtplugin

### DISCUSSION THREAD

**Discussion by tsujan** [lxqt/lxqt#2546] [General]
Dear LXQt dvelopers,

The method `QFont::toString()` has had a backward incompatible change in Qt6, such that the font set by the Qt6-based lxqt-config won't be readable by the Qt5-based lxqt-qtplugin. As a result, fonts couldn't be set for Qt5 apps when we release LXQt 2.0.0.

For example, Qt6 sets the font as `font="DejaVu Sans,9,-1,5,400,0,0,0,0,0,0,0,0,0,0,1"` in `~/.config/lxqt/lxqt.conf`, while Qt5 set the same font as `font="DejaVu Sans,9,-1,5,50,0,0,0,0,0"`, and the former can't be read by `QFont::fromString` in the Qt5-based lxqt-qtplugin.

The "detached" branch [fix_fontapply](https://github.com/lxqt/lxqt-qtplugin/tree/fix_fontapply) — kindly created by @stefonarch — includes a simple and backward compatible workaround (check the function `lxqtplatformtheme.cpp` → `fontFromString` in it). The workaround is made based on a similar workaround that I used in FeatherPad when it still supported both Qt6 and Qt5. 

I think I could add a signed tag to that branch and release lxqt-qtplugin 1.4.1 based on it, without affecting our current master. Since, as far as I know, we haven't done this kind of job before, I wanted to ask your opinion. I won't do it unless two LXQt members approve of it — of course, except for me and @stefonarch, who have tested it successfully.

@palinek, @yan12125, @luis-pereira,... ?

**Comment by yan12125** [comment]
Thank you both for taking care of this!

> I think I could add a signed tag to that branch and release lxqt-qtplugin 1.4.1 based on it

There might be more fixes to the old version in the future (ex: new build failures caused by stricter compilers). How about a more generic branch name like `1.x`? Also, I feel it a liitle bit strange to have a release tag on a branch called `fix_fontapply`.

> The "detached" branch [fix_fontapply](https://github.com/lxqt/lxqt-qtplugin/tree/fix_fontapply) — kindly created by @stefonarch — includes a simple and backward compatible workaround

In the long term, maybe it's better to define our own format and move away from Qt serialization/deserialization functions? There might be a similar issue when Qt 7 comes.

**Reply by tsujan** [reply]
> How about a more generic branch name like 1.x? 

Good idea! We could rename it as `1.4`.

> In the long term, maybe it's better to define our own format... There might be a similar issue when Qt 7 comes.

IMHO, it's better to stick with Qt and deal with such exceptional cases. Distancing ourselves from Qt could cause more troubles in the long run.

**Reply by stefonarch** [reply]
>We could rename it as 1.4

Agreeing.

**Reply by yan12125** [reply]
> IMHO, it's better to stick with Qt and deal with such exceptional cases. Distancing ourselves from Qt could cause more troubles in the long run.

Mind to share more details? QTermimal uses custom properties `fontFamily` and `fontSize`, and it seems working fine.

https://github.com/lxqt/qterminal/blob/1.4.0/src/properties.cpp#L74

**Reply by tsujan** [reply]
See https://gitlab.com/wireshark/wireshark/-/issues/18553, which says exactly what I said in my first comment: There's no problem in Qt6 when reading a font string created by Qt5, but Qt5 can't read the font string created by Qt6, which could result in wrong fonts in Qt5 apps after LXQt 2.2.0 is released. Hence the workaround.

lxqt-qtplugin 1.4.1 should be released before 2.0.0, which will be released around 17 April.

**Reply by yan12125** [reply]
I'm okay with releasing 1.4.1 based on the status quo. My comments are general discussions about how such an issue can be avoided when Qt 7 arrives.

I will look into the Wireshark issue when I find time.

**Reply by tsujan** [reply]
> how such an issue can be avoided when Qt 7 arrives.

We can't predict in which places future versions of Qt might have backward incompatible changes. So, the answer is we couldn't avoid such situations, but we could handle them after encountering them.

**Comment by tsujan** [comment]
`fix_fontapply` is renamed as `1.4`.

**Comment by tsujan** [comment]
Made the point release https://github.com/lxqt/lxqt-qtplugin/releases/tag/1.4.1.
